src/constructs/s3/migrating-bucket.test.ts

```diff
@@ -0,0 +1,26 @@
+import "../../utils/test/jest";
+import { simpleGuStackForTesting } from "../../utils/test";
+import { GuMigratingS3Bucket } from "./migrating-bucket";
+
+describe("The GuMigratingS3Bucket construct", () => {
+  it("should auto-generate the logicalId by default", () => {
```

Comment:
I don't think we'd ever want this construct to create a new logical id (since it is explicitly for _migrating_ existing buckets)? Perhaps we could require the prop (or throw) instead of allowing this to happen?

Reply:
Good point! I actually don't think `GuMigratingS3Bucket` is in line with our other constructs, which are typically "Guardian flavoured versions of AWS constructs".

A Guardian flavoured bucket would be a tagged bucket with a retain policy. What if `GuMigratingS3Bucket` became `GuS3Bucket`, setting these props?

Reply:
Updated now. The diff is tiny as a result! 🎉